core/common/src/test/java/alluxio/TestLoggerRule.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import org.apache.log4j.AppenderSkeleton;
+import org.apache.log4j.Logger;
+import org.apache.log4j.spi.LoggingEvent;
+import org.junit.After;
+import org.junit.Before;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestLoggerRule extends AbstractResourceRule{
+
+  public TestAppender mAppender;
+
+  @Before
+  public void before() {
+    mAppender = new TestAppender();
+    Logger.getRootLogger().addAppender(mAppender);
+  }
+
+  @After
+  public void after() {
+    Logger.getRootLogger().removeAppender(mAppender);
+  }
+
+  public class TestAppender extends AppenderSkeleton {
+
+    public List<LoggingEvent> mEvents = new ArrayList<LoggingEvent>();
+
+    public void close() { }
+
+    /**
+     * Determines whether string was logged.
+     */
+    public boolean wasLogged(String eventString) {
```

Comment:
can we also implement `wasLogged` on the `TestLoggerRule` object so that rule users can call `mLogger.wasLogged()` instead of `mLogger.mAppender.wasLogged`? Then we can make `mAppender` private

Reply:
I was considering this before but wasn't sure. I'll make the change.